src/task-controller.js

```diff
@@ -94,19 +104,15 @@ class TaskController extends AbortController {
       throw new TypeError(`Invalid task priority: '${priority}'`);
     }
 
-    /**
-     * @private
-     * @type {string}
-     */
-    this.priority_ = priority;
-
     /**
      * @private
      * @type {boolean}
      */
     this.isPriorityChanging_ = false;
 
-    makeTaskSignal(this);
+    // Morph the AbortSignal instance into a TaskSignal instance.
+    Object.setPrototypeOf(this.signal, TaskSignal.prototype);
```

Comment:
My only concern is if this causes slowness as warned here:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf

But can't say I know enough about `setPrototypeOf`, nor if this example is small enough, that it's OK here.

Reply:
I made an arbitrary test on jsbenchmark and it seems fine. I know it's not very accurate, but it should catch any big problems: https://jsbm.dev/uq67sfUiVx8bX

Reply:
Nice. Thanks for that.

@shaseley think we're good to merge then?

Reply:
Yep!